wgsl-struct-alignment-proposal.md

```diff
@@ -132,6 +132,43 @@ fn foo() -> vec2<f32> {
 }
 ```
 
+#### Interaction with atomics
+
+Atomics are coming. See https://github.com/gpuweb/gpuweb/issues/1360
+
+The widen-to-vec4 technique doesn't work directly if one of the fields being absorbed is atomic:
+```rust
+[[block]] struct Uniforms {
+    a : vec2<u32>;
+    b : atomic<u32>;
+    c : u32;
+};
+```
+
+Suppose we translate it like this:
+```rust
+[[block]] struct Uniforms {
+    a : vec4<u32>;
+};
+```
+
+Consider the Vulkan translation. 
+We should only touch the `b` component via atomic operations.
+We might introduce data races if we do non-atomic loads and stores on the whole vector or on what would be the `b` component. 
+
+The right solution is likely to scalarize the 4 elements of the vector:
```

Comment:
I assume sequences of scalars are contiguously laid out for all backends then?

Reply:
Yes, to my understanding.